azure/scope/managedmachinepool.go

```diff
@@ -224,6 +224,52 @@ func buildAgentPoolSpec(managedControlPlane *infrav1.AzureManagedControlPlane,
 		}
 	}
 
+	if managedMachinePool.Spec.LinuxOSConfig != nil {
+		agentPoolSpec.LinuxOSConfig = &agentpools.LinuxOSConfig{
+			SwapFileSizeMB:             managedMachinePool.Spec.LinuxOSConfig.SwapFileSizeMB,
+			TransparentHugePageDefrag:  (*string)(managedMachinePool.Spec.LinuxOSConfig.TransparentHugePageDefrag),
+			TransparentHugePageEnabled: (*string)(managedMachinePool.Spec.LinuxOSConfig.TransparentHugePageEnabled),
+		}
+		if managedMachinePool.Spec.LinuxOSConfig.Sysctls != nil {
+			var ipv4LocalPortRange string
+			if portRange := managedMachinePool.Spec.LinuxOSConfig.Sysctls.NetIpv4IPLocalPortRange; portRange != nil {
+				// NetIpv4IPLocalPortRange.First and NetIpv4IPLocalPortRange.Last are nil checked at webhook level
+				ipv4LocalPortRange = fmt.Sprintf("%d %d", *portRange.First, *portRange.Last)
```

Comment:
even if the webhooks check for nil, we should still use `pointers.StringDeref()` to be safe